### PR TITLE
Update DAGs

### PR DESCRIPTION
**Pull request by mpgreg**
Refactored DAGs to import from includes.
New version of weaviate provider had upsert and branch operator for check schema.
Add cohere rerank to schema
Moved env vars to .env
Added openlineage data source
Fixes to enable astro deploy
Added ASK_ASTRO_ENV var to enable the same code to run in all environments. Specify 'prod', 'dev', 'test', or 'local' in each astro deployment or local CLI.

### CODE REVIEW

File: airflow/requirements.txt [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 markdownify
 stackapi==0.2.0
-protobuf==3.20.1
-google-cloud-firestore
-openai
\ No newline at end of file
+protobuf==3.20.2
+google-cloud-firestore==2.12.0
+openai==0.28.1
+bs4==0.0.1
```
Comment (Lee-W) [inline airflow/requirements.txt:14]: Looks like we're removing it in this pr. do we still need it?

File: airflow/requirements.txt [hunk 2]
```diff
@@ -1,13 +1,15 @@
-/tmp/airflow_provider_weaviate-1.0.0-py3-none-any.whl
+/tmp/airflow_provider_weaviate-0.0.1-py3-none-any.whl
 apache-airflow-providers-github==2.3.1
 apache-airflow-providers-slack==7.3.2
 weaviate-client==3.22.0
 streamlit==1.25.0
-html2text
+html2text==2020.1.16
```
Comment (Lee-W) [inline airflow/requirements.txt:6]: May I know why do we want to pin these versions? If this is what we want, should we pin all the versions?

File: airflow/requirements.txt [hunk 3]
```diff
@@ -1,13 +1,15 @@
-/tmp/airflow_provider_weaviate-1.0.0-py3-none-any.whl
+/tmp/airflow_provider_weaviate-0.0.1-py3-none-any.whl
```
Comment (Lee-W) [inline airflow/requirements.txt:1]: May I know why do we want to install the package this way? Should we push this to astronomer PyPI?

File: airflow/include/split.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import pandas as pd
+from typing import List
+
```
Comment (Lee-W) [inline airflow/include/split.py:3]:
```suggestion
from __future__ import annotations

import pandas as pd
```

File: airflow/include/split.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+from airflow.decorators import task
```
Comment (Lee-W) [inline airflow/include/split.py:4]: It seems this is not used
```suggestion
```

File: airflow/include/split.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+from langchain.text_splitter import (
+    RecursiveCharacterTextSplitter,
+    Language,
+)
+from langchain.schema import Document
+
+# @task
```
Comment (Lee-W) [inline airflow/include/split.py:12]:
```suggestion
```

File: airflow/include/split.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+def split_markdown(df:pd.DataFrame):
```
Comment (Lee-W) [inline airflow/include/split.py:13]: I think we should introduce tools like ruff and black to this project before we spend time on reviewing style, error that can be found/fixed by these tools

File: airflow/requirements.txt (hunk 1 above)
Comment (mpgreg) [reply airflow/requirements.txt:14]: I think you will still need bs4, right?

File: airflow/requirements.txt (hunk 2 above)
Comment (mpgreg) [reply airflow/requirements.txt:6]: Feel free to pin all.

File: airflow/requirements.txt (hunk 3 above)
Comment (mpgreg) [reply airflow/requirements.txt:1]: These providers are not yet on pypi.  They will be replaced by llm-providers later.

**Review by sunank200 — COMMENTED** [review]
We should rebase this branch with this pre-commit branch in this PR: https://github.com/astronomer/ask-astro/pull/19. I am waiting on this PR for pre-commit error to be resolved before merging this https://github.com/astronomer/ask-astro/pull/19 to main as well

**Comment by kaxil** [comment]
@mpgreg I have rebased your PR on main branch (that contains pre-commit hooks) and resolved merged conflicts & fixed issues.

**Comment by kaxil** [comment]
Why are tests not running on this PR 🤔

**Comment by kaxil** [comment]
Had to change a CircleCI setting

image

**Pull request closed without merging**